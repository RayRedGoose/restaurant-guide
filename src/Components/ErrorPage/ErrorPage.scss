@import "../../assets/css/variables.scss";

.error-card {
  background-color: #fff;
  border-radius: 1rem;
  box-shadow: 0px 0px 12px rgba(23, 126, 137, 0.2);
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 3rem;
  width: 80%;
  margin: auto;
  img {
    width: 5rem;
    margin-right: 3rem;
  }
  h2 {
    color: $error-color;
  }
}
