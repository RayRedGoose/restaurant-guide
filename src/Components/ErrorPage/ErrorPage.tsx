import './ErrorPage.scss';
import React from 'react';
import errorImage from 'assets/images/error.svg';

interface Props {
  error: string
}

const ErrorPage: React.FC<Props> = ({error}) => {
  return (
    <section className="error-card">
      <img src={errorImage} alt="error"/>
      <h2>{error}</h2>
    </section>
  );
};

export default ErrorPage;
