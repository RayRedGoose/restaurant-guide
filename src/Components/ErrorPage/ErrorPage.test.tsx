import React from 'react';
import { shallow } from 'enzyme';
import ErrorPage from './ErrorPage';

it('should match the snapshot', () => {
  const wrapper = shallow(<ErrorPage error="Failed!" />);
  expect(wrapper).toMatchSnapshot();
});
