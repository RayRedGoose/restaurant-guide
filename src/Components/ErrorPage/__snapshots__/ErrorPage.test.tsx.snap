// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<section
  className="error-card"
>
  <img
    alt="error"
    src="error.svg"
  />
  <h2>
    Failed!
  </h2>
</section>
`;
